### PR TITLE
Add autolabeler

### PR DESCRIPTION
**Pull request by rofafor**
As release-drafter is catogorizing releases notes based on labels, it's nice to integrate optional labeling support as well. Check the README for configuration.
image
image

### CODE REVIEW

**Comment by jetersen** [comment]
Hmm an action already exists for this: https://github.com/actions/labeler
I wonder if you could use the trick I did for https://github.com/release-drafter/release-drafter/blob/master/action.js to get that as a probot?

**Comment by jetersen** [comment]
Hmm, I like where this is heading though and it does seem different enough from actions/labeler.

**Comment by rofafor** [comment]
The actions/labeler supports only file paths, but this one tracks branch naming (for those using gitflow naming scheme), PR titles and descriptions (fix/bug/etc. keywords, ticketing system ids).

File: index.js [hunk 1]
```diff
@@ -21,6 +22,105 @@ module.exports = (app, { getRouter }) => {
     })
   }
 
+  app.on(
+    [
+      'pull_request.opened',
+      'pull_request.reopened',
+      'pull_request.synchronize',
+    ],
```
Comment (jetersen) [inline index.js:30]: Have you tested that this also works for GitHub actions?

Comment (rofafor) [reply index.js:30]: Not yet

Comment (jetersen) [reply index.js:30]: Would you mind testing it? 😅

File: lib/template.js [hunk 2]
```diff
@@ -52,5 +52,33 @@ function validateReplacers({ context, replacers }) {
...
+          }),
+          title: autolabel.title.map((reg) => {
+            return toRegex(reg)
+          }),
+          body: autolabel.body.map((reg) => {
+            return toRegex(reg)
+          }),
+        }
+      } catch (e) {
+        console.log({
```
Comment (RomanovX) [inline lib/template.js:72]:
```suggestion
        log({
```

I think your habit here turned to console.log rather than the built in log

Comment (rofafor) [reply lib/template.js]: Yes. I'm forward-porting these from an old fork, so mistakes happen :)

File: index.js (hunk 1 above)
Comment (rofafor) [reply index.js:30]: Will do.

Comment (rofafor) [reply index.js:30]: Seems to working with my initial tests. I updated the README howto enable.

File: package.json [hunk 3]
```diff
@@ -25,6 +25,7 @@
     "cli-table3": "0.6.0",
     "compare-versions": "3.6.0",
     "escape-string-regexp": "4.0.0",
+    "ignore": "^5.1.8",
```
Comment (jetersen) [inline package.json:28]: Please set it fixed and remember to update the `yarn.lock`

```suggestion
    "ignore": "5.1.8",
```

**Review by jetersen — APPROVED** [review]
LGTM, one minor nitpick than we are ready to merge.

File: package.json (hunk 3 above)
Comment (rofafor) [reply package.json]: done

**Pull request merged**